### ISSUE TITLE
Heroku deployment

### ISSUE THREAD

**Issue by HackPoint** [ghpabs/angular2-seed-project#46]
```2015-12-24T20:44:08.055382+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=frozen-falls-9270.herokuapp.com request_id=a04a3ad5-fd53-453d-96f6-6d126297a330 fwd="87.69.63.87" dyno= connect= service= status=503 bytes=
2015-12-24T20:44:08.375652+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=frozen-falls-9270.herokuapp.com request_id=0948e9a3-66cc-4751-9b57-9fb39681f765 fwd="87.69.63.87" dyno= connect= service= status=503 bytes=
2015-12-24T20:44:22.238725+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=frozen-falls-9270.herokuapp.com request_id=7fb45c04-df3d-4ab5-8531-0103ff293819 fwd="87.69.63.87" dyno= connect= service= status=503 bytes=
2015-12-24T20:44:22.529780+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=frozen-falls-9270.herokuapp.com request_id=ce6d92f4-1334-445a-a28d-01d9677638eb fwd="87.69.63.87" dyno= connect= service= status=503 bytes=
2015-12-24T20:44:02.141545+00:00 app[web.1]:     at Function.Module._resolveFilename (module.js:338:15)
2015-12-24T20:44:02.141548+00:00 app[web.1]:     at startup (node.js:138:18)
2015-12-24T20:44:02.141548+00:00 app[web.1]:     at node.js:974:3```

**Comment by ghpabs** [comment]
I'm afraid this is out of the scope of this project. You might need to seek help with Heroku's support team.

**Comment by HackPoint** [comment]
I was, I'm not so familiar with the whole CLI usage I'm a Visual Studio IDE favorite. So if you could help me even to understand what's the issue is he it will be much appreciated.

**Comment by ghpabs** [comment]
That's right, in Heroku go to Settings, there you need to set the Config Var `NPM_CONFIG_PRODUCTION=false` so Heroku will install all dependencies.

Heroku sees the `package.json` and understands it's a NodeJS app. So it will run `npm install` and `npm start`. That's why there's a `gulp build serve` inside the `package.json`.

**Comment by HackPoint** [comment]
Event after I did that and installed everything successfully it's still not working.

**Comment by ghpabs** [comment]
Cool, I've noticed that the todos won't load under https protocol, change it to http for now.

Closing the issue by now alright?

**Comment by celikmustafa89** [comment]
THIS SOLUTION IS FOR GO LANG

When you deploy an app through **heroku**, it does **not allow** you to specify the **port number**.   
In other words, you can not specify your web service's **port** number as **8000** or something else, heroku decides the port number in runtime.   
so, you can not use the following code:

        log.Fatal(http.ListenAndServe(":8000", router))

What you can do is, getting the runtime port of heroku.   
In short, **just use the following code**:

        log.Fatal(http.ListenAndServe(":" + os.Getenv("PORT"), router))
